incubator/hnc/internal/reconcilers/object.go

```diff
@@ -431,7 +432,10 @@ func (r *ObjectReconciler) syncSource(ctx context.Context, log logr.Logger, src
 	ns := r.Forest.Get(src.GetNamespace())
 	ns.SetSourceObject(src.DeepCopy())
 
-	if !r.shouldPropagateSource(log, src) {
+	// We are only checking if the source itself is propagatable, but not to a specific destination.
+	// We will enqueue all the descendants if it's propagatable and decide whether to propagate when
+	// reconciling the descendants.
+	if !r.shouldPropagateSource(log, src, "") {
 		// If an object was *previously* propagated by HNC, it will already be in the forest, and all
```

Comment:
This comment is out of date now - can you chat with Yiqi to update it?

Reply:
Updated